### PR TITLE
Add random insurgent units generator

### PR DESCRIPTION
**Pull request by veteran29**
**When merged this pull request will:**
- title

Based on https://github.com/ArmaForces/Mods/pull/59

### CODE REVIEW

File: addons/insurgents/$PBOPREFIX$ [hunk 1]
```diff
@@ -0,0 +1 @@
+z\tac\addons\insurgents
```
Comment (jonpas) [inline addons/insurgents/$PBOPREFIX$]:
```suggestion
x\tac\addons\insurgents
```

File: addons/insurgents/CfgVehicles.hpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class CfgVehicles {
+    class I_G_Soldier_base_F;
+    class I_G_Soldier_F: I_G_Soldier_base_F {
+        class EventHandlers;
+    };
+
+    class GVAR(soldier_i): I_G_Soldier_F {
+
```
Comment (jonpas) [inline addons/insurgents/CfgVehicles.hpp]:
```suggestion
```

File: addons/insurgents/config.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        name = COMPONENT_NAME;
+        units[] = {
+            QGVAR(soldier_i),
+            QGVAR(soldier_o),
+            QGVAR(soldier_b)
+        };
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {
+            "tac_common"
```
Comment (jonpas) [inline addons/insurgents/config.cpp]:
```suggestion
            "tac_main"
```
There is no `tac_common`. :)

Comment (veteran29) [reply addons/insurgents/config.cpp]: woops, just went with find and replace, testing this in game right now if works.

File: addons/insurgents/functions/fnc_addWeaponAmmo.sqf [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#include "script_component.hpp"
+/*
+ * Author: vetran29
+ * Add random compatible ammo to unit weapon.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Weapon <STRING>
+ * 2: Primary muzzle ammo count, defaults to 6 <NUMBER>
```
Comment (jonpas) [inline addons/insurgents/functions/fnc_addWeaponAmmo.sqf]:
```suggestion
 * 2: Primary muzzle ammo count <NUMBER> (default: 6)
```

File: addons/insurgents/functions/fnc_addWeaponAmmo.sqf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * 3: Additional muzzles ammo count, defaults to 3 <NUMBER>
```
Comment (jonpas) [inline addons/insurgents/functions/fnc_addWeaponAmmo.sqf]:
```suggestion
 * 3: Additional muzzles ammo count <NUMBER> (default: 3)
```

File: addons/insurgents/CfgGroups.hpp [hunk 6]
```diff
@@ -0,0 +1,339 @@
+class CfgGroups {
+    // INDEPENDENT
+    class Indep {
+        class IND_G_F {
+            class Infantry {
+                class GVAR(IND_G_F_sentry) {
+                    faction = "IND_G_F";
+                    name = "Insurgent Sentry";
```
Comment (jonpas) [inline addons/insurgents/CfgGroups.hpp]: Stringtable?

File: addons/insurgents/CfgGroups.hpp [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+                    icon = "\A3\ui_f\data\map\markers\nato\n_inf.paa";
+                    side = 2;
+
+                    class Unit0 {
+                        position[] = {0,0,0};
```
Comment (jonpas) [inline addons/insurgents/CfgGroups.hpp]:
```suggestion
                        position[] = {0, 0, 0};
```
Rest of it as well.

File: addons/insurgents/CfgGroups.hpp [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+                    class Unit1 {
+                        position[] = {5,-5,0};
+                        rank = "PRIVATE";
+                        side = 2;
+                        vehicle = QGVAR(soldier_i);
+                    };
+                };
+                class GVAR(IND_G_F_fireteam) {
+                    faction = "IND_G_F";
+                    name = "Insrugent Fire Team";
```
Comment (TyroneMF) [inline addons/insurgents/CfgGroups.hpp]: Insurgent*

File: addons/insurgents/CfgGroups.hpp [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+                    class Unit1 {
+                        position[] = {5,-5,0};
+                        rank = "PRIVATE";
+                        side = 1;
+                        vehicle = QGVAR(soldier_b);
+                    };
+                };
+                class GVAR(BLU_G_F_fireteam) {
+                    faction = "BLU_G_F";
+                    name = "Insrugent Fire Team";
```
Comment (TyroneMF) [inline addons/insurgents/CfgGroups.hpp]:
```suggestion
                    name = "Insurgent Fire Team";
```

File: addons/insurgents/CfgGroups.hpp (hunk 8 above)
Comment (TyroneMF) [inline addons/insurgents/CfgGroups.hpp]:
```suggestion
                    name = "Insurgent Fire Team";
```

File: addons/insurgents/CfgGroups.hpp [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+                    class Unit1 {
+                        position[] = {5,-5,0};
+                        rank = "PRIVATE";
+                        side = 0;
+                        vehicle = QGVAR(soldier_o);
+                    };
+                };
+                class GVAR(OPF_G_F_fireteam) {
+                    faction = "OPF_G_F";
+                    name = "Insrugent Fire Team";
```
Comment (TyroneMF) [inline addons/insurgents/CfgGroups.hpp]:
```suggestion
                    name = "Insurgent Fire Team";
```

**Comment by jonpas** [comment]
If you go to Files tab, you can merge all review comments in one. :)

**Comment by veteran29** [comment]
CRs applied.

**Comment by TyroneMF** [comment]
> I know, you're doing squash merges anyway arent you? :P
My poor emails!

File: addons/insurgents/CfgVehicles.hpp [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        headgearProbability = 0;
+        weapons[] = {"Throw", "Put"};
+        respawnWeapons[] = {"Throw", "Put"};
+        magazines[] = {};
+        respawnMagazines[] = {};
+        linkedItems[] = {"ItemMap", "ItemCompass", "ItemWatch", "ItemRadio"};
+        respawnlinkedItems[] = {"ItemMap", "ItemCompass", "ItemWatch", "ItemRadio"};
+
+        class EventHandlers: EventHandlers {
+            init = "";
```
Comment (jonpas) [inline addons/insurgents/CfgVehicles.hpp]: Will this not break XEH?

Comment (veteran29) [reply addons/insurgents/CfgVehicles.hpp]: XEH handlers are in subclass `CBA_Extended_EventHandlers` which should be inherited.

**Pull request merged**